src/vulkan/command_buffer.cc

```diff
@@ -117,6 +118,15 @@ Result CommandBuffer::SubmitAndReset(uint32_t timeout_ms) {
     return Result("Vulkan::Calling vkWaitForFences Fail (" + result_str + ")");
   }
 
+    /*
+  google/vulkan-performance-layers requires a call to vkDeviceWaitIdle or
+  vkQueueWaitIdle in order to report the information. Since we want to be
+  able to use that layer in conjunction with Amber we need to somehow
+  communicate that the Amber script has completed.
+  */
+  if (pipeline_runtime_layer_enabled)
+    device_->GetPtrs()->vkQueueWaitIdle(device_->GetVkQueue());
```

Comment:
Why don't we run this all of the time? Does it hurt?  It seems like this adds a lot of code just to pass a flag.